Core/src/org/sleuthkit/autopsy/geolocation/MapWaypoint.java

```diff
@@ -64,6 +67,18 @@
     private static final Logger logger = Logger.getLogger(MapWaypoint.class.getName());
     private final static String HTML_PROP_FORMAT = "<b>%s: </b>%s<br>";
     static private final SimpleDateFormat DATE_FORMAT = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ssX", Locale.US);
+    
+    private static final Map<Integer, Color> artifactTypesToColors = new HashMap();
+    
+    static {
+        artifactTypesToColors.put(BlackboardArtifact.ARTIFACT_TYPE.TSK_GPS_BOOKMARK.getTypeID(), Color.BLUE);
+        artifactTypesToColors.put(BlackboardArtifact.ARTIFACT_TYPE.TSK_GPS_LAST_KNOWN_LOCATION.getTypeID(), Color.RED);
+        artifactTypesToColors.put(BlackboardArtifact.ARTIFACT_TYPE.TSK_GPS_ROUTE.getTypeID(), Color.YELLOW);
+        artifactTypesToColors.put(BlackboardArtifact.ARTIFACT_TYPE.TSK_GPS_SEARCH.getTypeID(), Color.GREEN);
+        artifactTypesToColors.put(BlackboardArtifact.ARTIFACT_TYPE.TSK_GPS_TRACK.getTypeID(), Color.ORANGE);
+        artifactTypesToColors.put(BlackboardArtifact.ARTIFACT_TYPE.TSK_GPS_TRACKPOINT.getTypeID(), Color.ORANGE);
+        artifactTypesToColors.put(BlackboardArtifact.ARTIFACT_TYPE.TSK_METADATA_EXIF.getTypeID(), Color.CYAN);
+    }
```

Comment:
Why put the colors here instead of in Waypoint?  I would have added a getColor to Waypoint and have the subclass set the color if they want it to be something other than the default.

Reply:
I don't think Waypoint, which seems to be a data representation of a waypoint, should be storing information about its color, which is more of a visual feature and has nothing to do with the data inside the waypoint object.

Reply:
@rcordovano and I were talking about this the other day.  I think the Waypoint class (and its subclasses) will need to become a bit more than they are.  Adding a color property to them does not seem out of line as it is a property of the Waypoint.  I would like to see the color go there so that we can avoid this type checking.

Reply:
@ethteck and @kellykelly3, working from the premise that I tentatively stated in another comment, i.e., that MapWaypoint is a view class wrapped around a Waypoint model class, it makes sense to me to put color properties here in the view class instead of in the data model class. It is my recollection that Waypoint imposes a data model on the weakly typed BlackboardArtifact construct beneath it, which is the justification for the class. I certainly am a fan of separating the model and the view. 

Regarding the future of the Waypoint class, in the cited conversation I was wondering about how the fact that the refactoring Brian requested at the SleuthKit level has given us a new concept of a Waypoint data model class. Specifically, a waypoint as a property of a route. At the time, the meaning of Waypoint in the context of the geolocation packages was no longer loaded into my brain, so I was starting to question whether or not TSK Waypoints could take the place of Autopsy geolocation Waypoints. That notion is not surviving the light of day. Even the cursory glance I have taken at this Autopsy code shows that TSK Waypoints and Autopsy Waypoints are orthogonal, despite the unfortunate similarity in their names. Like I said, a TSK Waypoint is a component of a route and exists to support Java serialization, while an Autopsy Waypoint imposes a welcome layer of abstraction over a family of related artifacts that have no mechanism for expressing that homogeneity in terms of the geolocation data display data model. I think that MapWaypoint and Waypopint are looking secure in their respective roles and are in accord with the Single Responsibility Principle (SRP).  So Kelly, I think what we hammered out together when this feature first appeared is holding up.